src/tui.rs

```diff
@@ -734,37 +1412,143 @@ fn ui(f: &mut Frame, app: &App) {
+                .style(Style::default().fg(Color::Gray))
+                .bounds([min_week, max_week])
+                .labels(vec![
+                    Line::from("0"),
+                    Line::from(format!("{:.0}", max_week * 0.25)),
+                    Line::from(format!("{:.0}", max_week * 0.5)),
+                    Line::from(format!("{:.0}", max_week * 0.75)),
+                    Line::from(format!("{:.0}", max_week)),
+                ])
+        )
+        .y_axis(
+            Axis::default()
+                .title(app.currency.as_str())
+                .style(Style::default().fg(Color::Gray))
+                .bounds([y_min, y_max])
+                .labels(vec![
+                    Line::from(format!("{:.0}", y_min)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.25)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.5)),
+                    Line::from(format!("{:.0}", (y_min + y_max) * 0.75)),
+                    Line::from(format!("{:.0}", y_max)),
+                ])
+        );
 
-    f.render_widget(barchart, area);
+    f.render_widget(chart, area);
+}
+
+fn render_pie_chart(f: &mut Frame, area: Rect, portfolio: &Portfolio) {
+    let colors = [
+        Color::Red, Color::Green, Color::Blue, Color::Yellow, 
+        Color::Magenta, Color::Cyan, Color::White, Color::LightRed,
+    ];
+    
+    let positions: Vec<_> = portfolio.positions.iter().take(6).collect(); // Limit to 6 for horizontal display
+    let mut pie_lines = Vec::new();
+    
+    // Create compact horizontal bars with embedded labels
+    let mut chart_lines = Vec::new();
+    
+    for (i, position) in positions.iter().enumerate() {
+        let name = position.get_name();
+        let percentage = (position.get_balance() / portfolio.get_total_value()) * 100.0;
+        let color = colors[i % colors.len()];
+        
+        // Create horizontal bar (max 30 characters wide)
+        let bar_width = ((percentage / 100.0) * 30.0) as usize;
+        let bar_width = bar_width.max(1).min(30);
+        
+        // Truncate name to fit in available space
+        let display_name = if name.len() > 12 { &name[..12] } else { name };
+        
+        let mut line_spans = Vec::new();
+        line_spans.push(Span::styled("● ", Style::default().fg(color)));
+        line_spans.push(Span::styled(format!("{:<12}", display_name), Style::default().fg(Color::White)));
+        line_spans.push(Span::styled(format!("{:>6.1}% ", percentage), Style::default().fg(color)));
+        line_spans.push(Span::styled("█".repeat(bar_width), Style::default().fg(color)));
+        
+        chart_lines.push(Line::from(line_spans));
+    }
+    
+    pie_lines.extend(chart_lines);
+    
+    if portfolio.positions.len() > 6 {
+        pie_lines.push(Line::from(""));
+        pie_lines.push(Line::from(vec![
+            Span::styled(format!("... and {} more positions", portfolio.positions.len() - 6), Style::default().fg(Color::Gray))
```

Comment:
I think it's fine to cut off after a certain length, but then please show the largest positions on top. Currently, they are ordered by file position.

<img width="238" height="146" alt="Image" src="https://github.com/user-attachments/assets/86ecae63-2ddb-4524-9baf-a509a2782524" />